rackcity/views/it_instance_views.py

```diff
@@ -205,3 +208,72 @@ def is_location_full(rack_id, instance_elevation, instance_height):
             if occupied_location in new_instance_location_range:
                 return True
     return False
+
+
+def get_sort_arguments(data):
+    sort_args = []
+    if 'sort_by' in data:
+        sort_by = data['sort_by']
+        for sort in sort_by:
+            if ('field' not in sort) or ('ascending' not in sort):
+                raise Exception("Must specify 'field' and 'ascending' fields.")
+            if not isinstance(sort['field'], str):
+                raise Exception("Field 'field' must be of type string.")
+            if not isinstance(sort['ascending'], bool):
+                raise Exception("Field 'ascending' must be of type bool.")
+            field_name = sort['field']
+            order = "-" if not bool(sort['ascending']) else ""
+            sort_args.append(order + field_name)
+    return sort_args
+
+
+def get_filter_arguments(data):
+    filter_args = {}
+    if 'filters' in data:
+        filters = data['filters']
+        for filter in filters:
+
+            if (
+                ('field' not in filter)
+                or ('filter_type' not in filter)
+                or ('filter' not in filter)
+            ):
+                raise Exception(
+                    "Must specify 'field', 'filter_type', and 'filter' fields."
+                )
+            if not isinstance(filter['field'], str):
+                raise Exception("Field 'field' must be of type string.")
+            if not isinstance(filter['filter_type'], str):
+                raise Exception("Field 'filter_type' must be of type string.")
+            if not isinstance(filter['filter'], dict):
+                raise Exception("Field 'filter' must be of type dict.")
+
+            filter_field = filter['field']
+            filter_type = filter['filter_type']
+            filter_dict = filter['filter']
+
+            if filter_type == 'text':
+                if filter_dict['match_type'] == 'exact':
+                    filter_args['{0}'.format(filter_field)] = \
+                        filter_dict['value']
+                elif filter_dict['match_type'] == 'contains':
+                    filter_args['{0}__icontains'.format(filter_field)] = \
+                        filter_dict['value']
+
+            elif filter_type == 'numeric':
+                range_value = (
+                    int(filter_dict['min']),
+                    int(filter_dict['max'])
+                )
+                filter_args['{0}__range'.format(filter_field)] = range_value
+
+            elif filter_type == 'rack_range':
```

Comment:
Do we have a ticket to add this in later?

Reply:
just made one!